server/controllers/stock/index.js

```diff
@@ -622,15 +622,67 @@ async function listLotsDepot(req, res, next) {
  * @todo process stock alert, rupture of stock
  * @todo prevision for purchase
  */
-function listInventoryDepot(req, res, next) {
+async function listInventoryDepot(req, res, next) {
   const params = req.query;
   const monthAverageConsumption = req.session.enterprise.settings.month_average_consumption;
   const enableDailyConsumption = req.session.enterprise.settings.enable_daily_consumption;
 
-  core.getInventoryQuantityAndConsumption(params, monthAverageConsumption, enableDailyConsumption)
-    .then((rows) => res.status(200).json(rows))
-    .catch(next)
-    .done();
+  try {
+    const inventoriesParameters = [params, monthAverageConsumption, enableDailyConsumption];
+    const inventories = await core.getInventoryQuantityAndConsumption(...inventoriesParameters);
+    const lots = await core.getLotsDepot(null, params);
+
+    for (let i = 0; i < inventories.length; i++) {
+      let hasRiskyLots = false;
+      let hasExpiredLots = false;
+      let hasNearExpireLots = false;
+
+      let riskyLotsQuantity = 0;
+      let expiredLotsQuantity = 0;
+      let nearExpireLotsQuantity = 0;
+
+      for (let j = 0; j < lots.length; j++) {
+        const hasSameDepot = lots[j].depot_uuid === inventories[i].depot_uuid;
+        const hasSameInventory = lots[j].inventory_uuid === inventories[i].inventory_uuid;
+        if (hasSameDepot && hasSameInventory) {
+          const lot = lots[j];
+          if (lot.quantity <= 0) {
```

Comment:
Is this equivalent to @jmcameron's code on the client?  If so, can you make it into a function of the same name here in the file?  That would make it easier to update/merge.